### PR TITLE
query moved to redux

### PR DESCRIPTION
**Pull request by mattydono**
Moved query to redux in order to remove emit from search

### CODE REVIEW

File: src/features/search/components/stockList.tsx [hunk 1]
```diff
@@ -63,21 +63,20 @@ type _Stock = {
...
+export const StockList = memo<StockListProps>(({inputSelect, setSelectedStock, dropSelect}) => {
 
     const dispatch = useDispatch()
     const stockList = useSelector((state: AppState) => state.search.stockList)
 
     const onStockClick = (stock: _Stock) => {
         const stockSymbol = stock.symbol
         const stockName = stock.name
-        setQuery(`${stockName} (${stockSymbol})`)
+        dispatch(updateQuery(`${stockName} (${stockSymbol})`))
```
Comment (danielabbasi) [inline src/features/search/components/stockList.tsx:79]: Q: If I understand correctly, this action is supposed to send the search input on every key press to the server. Can you explain why you need to send the stock name as well?

Comment (mattydono) [reply src/features/search/components/stockList.tsx:79]: So here query is being updated for label overlay purposes.

The overlay has to = query in order to match match on/off of focus.

File: src/features/search/redux/middleware.ts [hunk 2]
```diff
@@ -17,6 +17,11 @@ export const searchMiddleware = ({ socketService }: MiddlewareDependencies): Mid
                 socket.emit('chart', [action.payload, range])
             }
 
+            if (isActionOf(updateQuery, action)) {
+                if (action.payload === '') dispatch(updateStockList([]))
+                socket.emit('search', action.payload)
```
Comment (zhouyuhe) [inline src/features/search/redux/middleware.ts:22]: why `action.payload === '' `?

Comment (mattydono) [reply src/features/search/redux/middleware.ts:22]: so when they delete whatever they have entered in query (input) stocklist state is changed to [] as we know nothing = nothing

**Review by zhouyuhe — APPROVED** [review]

**Pull request merged**